web/src/recent_view_ui.ts

```diff
@@ -1196,19 +1191,23 @@ function recenter_focus_if_off_screen(): void {
     }
 }
 
-function is_scroll_position_for_render(scroll_container: HTMLElement): boolean {
-    const table_bottom_margin = 100; // Extra margin at the bottom of table.
-    const table_row_height = 50;
-    return (
-        scroll_container.scrollTop +
-            scroll_container.clientHeight +
-            table_bottom_margin +
-            table_row_height >
-        scroll_container.scrollHeight
-    );
+function is_scroll_position_for_render(): boolean {
+    // We render new rows if user has already scrolled 2 / 3 of rendered row.
+    const scroll_position = window.scrollY;
+    const scroll_height = window.innerHeight;
```

Comment:
I think `window_height` might be more appropriate as a name here.

Reply:
Yeah, changed.